R/html-tweak.R

```diff
@@ -324,6 +318,81 @@ badges_extract_text <- function(x) {
   xml <- xml2::read_html(x)
   badges_extract(xml)
 }
+
+activate_navbar <- function(html, path, pkg) {
+
+  html_navbar <- xml2::xml_find_first(html, ".//div[contains(@class, 'navbar')]")
+  nav_items <- xml2::xml_find_all(html_navbar,".//li[contains(@class, 'nav-item')]")
+
+  navbar_haystack <- navbar_links_haystack(nav_items, pkg)
+  path <- remove_useless_parts(path, pkg = pkg)
+
+  separate_path <- function(link) {
+    strsplit(link, "/")[[1]]
+  }
+
+  get_similarity <- function(stalk, needle) {
+    needle <- separate_path(needle)
+    stalk <- separate_path(stalk)
+
+    # Active item can't be more precise than current path/needle
+    if (length(stalk) > length(needle)) {
+      return(0)
+    }
+
+    # Active item can however be less precise than current path/needle
+    length(stalk) <- length(needle)
```

Comment:
if the stalk is shorter than the needle here we add NA at the end which makes it possible to compare stalk and needle.

Reply:
It would be slightly simpler to pad with `""`, then you wouldn't need the `na.rm` below.

Reply:
👍